src/Database/LSMTree/Internal/Merge.hs

```diff
@@ -199,13 +179,16 @@ close fs Merge {..} = do
 -- run they came from. This is important for resolving multiple entries with the
 -- same key into one.
 --
--- Construct with 'newReaders', then keep calling 'getFromReaders'.
--- If aborting early, remember to call 'closeReaders'!
+-- Construct with 'readersNew', then keep calling 'readersPop'.
+-- If aborting early, remember to call 'readersClose'!
 data Readers fhandle = Readers {
-      readersHeap     :: !(Heap.MutableHeap RealWorld (ReadCtx fhandle))
+      readersHeap :: !(Heap.MutableHeap RealWorld (ReadCtx fhandle))
       -- | Since there is always one reader outside of the heap, we need to
-      -- store it separately.
-    , readersLastRead :: !(IORef (RunReader fhandle, ReaderNumber))
+      -- store it separately. This also contains the next k\/op to yield, unless
+      -- all readers are drained, i.e. both:
+      -- 1. the reader it contains is empty
+      -- 2. the heap is empty
```

Comment:
I think 2. implies 1., and the comment slightly confused me

Reply:
It's not the same! 1. talks about the reader inside the `ReadCtx` that is stored separately from the heap. I understand the confusion though and clarified the comment.

Reply:
Ah, since the reader in 1. is outside of the heap?